File: Esame_22/PYTHON/Teoria.py
from Corso import Corso # importo Corso dal modulo Corso

class Teoria(Corso): # classe Teoria che eredita da corso

    # COSTRUTTORE, esso:
    def __init__(self, codiceCorso, nomeCorso, docente, codiceAula, oreSettimanali, orePerLezione):
        super().__init__(codiceCorso, nomeCorso, docente) # chiama il costruttore della superclasse
        self._codiceAula = codiceAula   # ed assegna agli attributi _
        self._oreSettimanali = oreSettimanali
        self._orePerLezione = orePerLezione

    # METODI:
    def toString(self):
        return super().toString() + "teoria\t" + self._codiceAula + "\t" + \
               self._oreSettimanali + "\t" + self._orePerLezione + "\t-\t-\t-\t"

File: Esame_22/JAVA/Teoria.java
import java.util.*;

public class Teoria extends Corso {

	// Attributi
	private int codiceAula;
	private int oreSettimanali;
	private double orePerLezione;

	// Costruttore
	public Teoria(int codiceCorso, String nomeCorso, String docente, int codiceAula, int oreSettimanali,
			double orePerLezione) {
		super(codiceCorso, nomeCorso, docente);
		this.codiceAula = codiceAula;
		this.oreSettimanali = oreSettimanali;
		this.orePerLezione = orePerLezione;
	}

	// Metodi
	// PUNTO 3: toString
	public String toString() {
		return super.toString() + "teoria" + "\t" + this.codiceAula + "\t" + this.oreSettimanali + "\t"
				+ this.orePerLezione + "\t-\t-\t-\t";
	}
}
